scripts/update-message-registry.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+
+import argparse
+from dataclasses import astuple, dataclass
+from pathlib import Path
+from typing import Sequence
+
+import jinja2
+import yaml
+
+
+@dataclass(order=True)
+class Version:
+    """A semantic version number: MAJOR.MINOR.PATCH."""
+
+    UNKNOWN_VERSION = "UNKNOWN"
+    DEFAULT_VERSION = "1.0"  # [TODO] some initial version?
+
+    major: int
+    minor: int
+    patch: int = 0
+
+    def __str__(self):
+        return ".".join(map(str, astuple(self)))
+
+    @classmethod
+    def parse(cls, raw_version: str) -> Version:
+        if raw_version == cls.UNKNOWN_VERSION:
+            raw_version = cls.DEFAULT_VERSION
+        version = raw_version.split(".")
+        assert 2 <= len(version) <= 3 and all(
+            n.isdecimal() for n in version
+        ), raw_version
+        return Version(*map(int, version))
+
+
+@dataclass
+class Example:
+    """An example in a message entry."""
+
+    name: str
+    description: str
+    before: str | None
+    after: str | None
+
+    @classmethod
+    def parse(cls, entry) -> Example:
+        name = entry.get("name")
+        assert name, entry
+        description = entry.get("description")
+        assert description
+        before = entry.get("before")
+        after = entry.get("after")
+        # Either none or both of them
+        assert not (bool(before) ^ bool(after))
+        return Example(name=name, description=description, before=before, after=after)
+
+
+@dataclass
+class Entry:
+    """An xkbcommon message entry in the message registry"""
+
+    VALID_TYPES = ("warning", "error")
+
+    code: int
+    """A unique strictly positive integer identifier"""
+    id: str
+    """A unique short human-readable string identifier"""
+    type: str
+    """Log level of the message"""
+    description: str
+    """A short description of the meaning of the message"""
+    details: str
+    """A long description of the meaning of the message"""
+    added: Version
+    """Version of xkbcommon the message has been added"""
+    removed: Version | None
```

Comment:
I think this requires Python 3.9 which *may* be a problem (and the walrus requires 3.8 iirc).  I'm having a deja-vu here but github is not fond of easily showing me outdated conversations, so apologies if we already discussed this here somewhere.

Reply:
1. this is a script aimed for _developers only_, so it should not be distributed and there is no issue with CI.
2. Python 3.9 has been released in 2020. It is not exactly recent and it brings lots of benefits.

I am not against lowering Python bounds, but I do not see why we should held back on old Python versions for _developer scripts_. Recent versions bring lots of nice enhancements and it is quite easy to have multiple versions of Python installed in parallel.

I agree Python version must be carefully chosen for things such as build scripts, running on CI or on user side, but here this is _not_ the case.

We should agree and indicate the minimum requirements to work on this repo.